cpp/src/arrow/flight/test_definitions.cc

```diff
@@ -145,6 +167,13 @@ void DataTest::CheckDoGet(
   ASSERT_OK_AND_ASSIGN(auto info, client_->GetFlightInfo(descr));
   check_endpoints(info->endpoints());
 
+  if (supports_async()) {
+    auto listener = std::make_shared<GetFlightInfoListener>();
+    client_->GetFlightInfo(descr, listener);
+    ASSERT_FINISHES_OK(listener->future);
```

Comment:
This begs the question: do we want to a client `GetFlightInfoAsync` overload that returns a `Future` directly?

I presume that, at some point, `AsyncListener` may grow other capabilities, but simple uses might be content with getting a `Future`...

Reply:
I was considering that; gRPC actually does that, basically (there is an overload that just takes a `std::function` for unary-unary calls like this).